### DISCUSSION TITLE
How do I colorize some YAML code without affecting background colors?

### DISCUSSION THREAD

**Discussion by ssbarnea** [Textualize/rich#990] [Q&A] [answered]
I currently use [`Syntax(text, 'yaml')`](https://github.com/ansible-community/ansible-lint/blob/master/src/ansiblelint/color.py#L46) in order to render some YAML files nicely on screen. Still I observed that this is changing the background color, something that is undesired, especially as some terminal users use darn-on-white schemes.

![](https://sbarnea.com/ss/Screen-Shot-2021-02-05-12-58-29.30.png)

I tried giving a theme parameter like `theme='ansi_dark'` and this produced ok results, for normal terminals, until one of the users send me this:
![](https://usercontent.irccloud-cdn.com/file/TjTqiOuv/ansible-lint-output.png)

Is there a way to configure rich in such way that it would render yaml without affecting background?

**Comment by willmcgugan** [comment, marked as answer]
In that last screenshot I'm pretty sure you are using the default them of "monokai" which has a non-transparent background color. Use "ansi_light" or "ansi_dark" for a transparent background.

You can force the background color to be transparent with `background_color="default"` although that may produce unreadable text for some terminals.

**Reply by ssbarnea** [reply]
Indeed his screenshot is made with an older version, i know that as it also has the border which I removed since.

Knowing that these two themes have transparent background is very useful, making them prime candidates for us.

My only question is: can't we assure that the default theme uses a transparent background on rich?

For the moment I will manually switch to ansi_dark but it will be awesome if default one would be transparent.

**Reply by willmcgugan** [reply]
Monokai is the default theme because I think it looks the best. Obviously a personal choice. It existed prior to adding the ansi_* themes. I see no reason to change the default now, since its trivial to change.
